### PR TITLE
Enable fastpath for in operator

### PR DESCRIPTION
**Pull request by meg-gupta**
This was originally developed by Etienne Baudoux for javascript implementation of indexOf.
Merging this now with bug fix.

### CODE REVIEW

**Comment by meg-gupta** [comment]
@Cellule @MikeHolman - Most of this change is from Etienne's indexof work.

File: lib/Backend/Lower.cpp [hunk 1]
```diff
@@ -19398,6 +19399,101 @@ Lowerer::GenerateFastInlineRegExpExec(IR::Instr * instr)
...
+
+    vtableOpnd = LoadVTableValueOpnd(instr, VTableValue::VtableNativeFloatArray);
+    InsertCompareBranch(
+        IR::IndirOpnd::New(arrayOpnd, 0, TyMachPtr, instr->m_func),
+        vtableOpnd,
+        Js::OpCode::BrNeq_A,
+        helperLabel,
+        instr);
+
+    instr->InsertBefore(isArrayLabel);
```
Comment (agarwal-sandeep) [inline lib/Backend/Lower.cpp]: GenerateArrayTest does most of the work you are doing here, can't GenerateArrayTest be used instead?

Comment (meg-gupta) [reply lib/Backend/Lower.cpp]: It looks much more complicated than the tests done here, and doesn't have all the checks needed as well. I think having a new function is much more easy to read.

**Pull request merged**